src/include/buffer.h

```diff
@@ -378,6 +378,88 @@ namespace buffer CEPH_BUFFER_API {
       }
     };
 
+    class appender {
+    protected:
+      list *pls;
+      size_t size;
+      ptr bp;
+      char *pos, *end;
+
+    public:
+      appender(bufferlist *l, size_t s) : pls(l), size(s) {
+	prepare_buffer();
+      }
+      appender(appender&& o)
```

Comment:
move constructor.  not sure if this is the right way to implement it, but 
we do want to disallow copying and only allow moves so that we have an 
accurate position to flush() when the destructor is called.